File: content/05/code/dotnet/21--step21-tidy-up-network-logic/Shouty/Shouty/Network.cs
using System;
using System.Collections.Generic;

namespace Shouty
{
    public class Network
    {
        private readonly List<Person> listeners = new List<Person>();
        private readonly int range;

        public Network(int range)
        {
            this.range = range;
        }

        public virtual void Subscribe(Person person)
        {
            listeners.Add(person);
        }

        public virtual void Broadcast(string message, int shouterLocation)
        {
            foreach (var listener in listeners)
            {
                var withinRange = Math.Abs(listener.Location - shouterLocation) <= range;
                var shortEnough = message.Length <= 180;
                if (withinRange && shortEnough)
                    listener.Hear(message);
            }
        }
    }
}

File: content/bdd-with-cucumber/06/code/java/16-disable-and-silence-cucumber-reports/shouty/src/main/java/shouty/Network.java
package shouty;

import java.util.ArrayList;
import java.util.List;

public class Network {
    private final List<Person> listeners = new ArrayList<Person>();
    private final int range;

    public Network(int range) {
        this.range = range;
    }

    public void subscribe(Person person) {
        listeners.add(person);
    }

    public void broadcast(String message, int shouterLocation) {
        for (Person listener : listeners) {
            boolean withinRange = (Math.abs(listener.getLocation() - shouterLocation) <= range);
            boolean shortEnough = (message.length() <= 180);
            if (withinRange && shortEnough) {
                listener.hear(message);
            }
        }
    }
}